v8.go

```diff
@@ -77,14 +77,19 @@ type Snapshot struct{ data C.StartupData }
 
 func newSnapshot(data C.StartupData) *Snapshot {
 	s := &Snapshot{data}
-	runtime.SetFinalizer(s, (*Snapshot).release)
```

Comment:
Yep.  Move the existing Context content to an internal contextData struct and use that for all of the self-referential stuff.  Then create a new Context struct that just holds a *contextData.  Same with Isolate.  Values should point to contextData, so we should have a strict ordering of:

`Isolate -> Context[s] -> Value[s]`